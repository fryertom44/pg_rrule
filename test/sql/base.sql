\set ECHO none
BEGIN;
\i sql/pg_rrule.sql
\set ECHO all

SET TIME ZONE 'GMT';
SET DATESTYLE TO 'ISO';

SELECT 'FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z'::rrule;

SELECT get_byday('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=MO,TH,SU'::rrule);

SELECT get_freq('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z'::rrule);

SELECT * FROM
    unnest(
        get_occurrences('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=SA;BYHOUR=10;BYMINUTE=51;BYSECOND=2'::rrule,
            '2019-12-07 10:51:02+00'::timestamp with time zone)
    );

SELECT * FROM
    unnest(
        get_occurrences('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=SA;BYHOUR=10;BYMINUTE=51;BYSECOND=2'::rrule,
            '2019-12-07 10:51:02'::timestamp)
    );

ROLLBACK;
