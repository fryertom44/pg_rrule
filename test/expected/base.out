\set ECHO none
SET TIME ZONE 'GMT';
SET DATESTYLE TO 'ISO';
SELECT 'FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z'::rrule;
               rrule                
------------------------------------
 FREQ=WEEKLY;UNTIL=20200101T045102Z
(1 row)

SELECT get_byday('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=MO,TH,SU'::rrule);
 get_byday 
-----------
 {2,5,1}
(1 row)

SELECT get_freq('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z'::rrule);
 get_freq 
----------
 WEEKLY
(1 row)

SELECT * FROM
    unnest(
        get_occurrences('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=SA;BYHOUR=10;BYMINUTE=51;BYSECOND=2'::rrule,
            '2019-12-07 10:51:02+00'::timestamp with time zone)
    );
         unnest         
------------------------
 2019-12-07 10:51:02+00
 2019-12-14 10:51:02+00
 2019-12-21 10:51:02+00
 2019-12-28 10:51:02+00
(4 rows)

SELECT * FROM
    unnest(
        get_occurrences('FREQ=WEEKLY;INTERVAL=1;WKST=MO;UNTIL=20200101T045102Z;BYDAY=SA;BYHOUR=10;BYMINUTE=51;BYSECOND=2'::rrule,
            '2019-12-07 10:51:02'::timestamp)
    );
       unnest        
---------------------
 2019-12-07 10:51:02
 2019-12-14 10:51:02
 2019-12-21 10:51:02
 2019-12-28 10:51:02
(4 rows)

ROLLBACK;
